operators/test/e2e/stack/checks_k8s.go

```diff
@@ -317,3 +322,38 @@ func CheckESPassword(stack Builder, k *helpers.K8sHelper) helpers.TestStep {
 		}),
 	}
 }
+
+// KibanaDoReq executes an HTTP request against a Kibana instance.
+func KibanaDoReq(k *helpers.K8sHelper, stack Builder, method string, uri string, body []byte) ([]byte, error) {
+	password, err := k.GetElasticPassword(stack.Elasticsearch.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	u, err := url.Parse(fmt.Sprintf("http://%s.%s.svc:5601", kbname.HTTPService(stack.Kibana.Name), stack.Kibana.Namespace))
+	if err != nil {
+		return nil, err
+	}
+
+	u.Path = path.Join(u.Path, uri)
+	req, err := http.NewRequest(method, u.String(), bytes.NewBuffer(body))
+	if err != nil {
+		return nil, err
+	}
+
+	req.SetBasicAuth("elastic", password)
+	req.Header.Set("kbn-version", stack.Kibana.Spec.Version)
```

Comment:
do we need that?

Reply:
Yes. Kibana expects the header `kbn-version` to be passed on non GET requests, it's used to protect against xsrf attacks (https://github.com/elastic/kibana/pull/5587).
Otherwise, you get `{"statusCode":400,"error":"Bad Request","message":"Request must contain a kbn-xsrf header."}`.
I will add a comment.

Reply:
Yes. Kibana expects the header `kbn-version` to be passed on non GET requests, it's used to protect against xsrf attacks (https://github.com/elastic/kibana/pull/5587).
Otherwise, you get `{"statusCode":400,"error":"Bad Request","message":"Request must contain a kbn-xsrf header."}`.
I will add a comment.